### ISSUE TITLE
Custom ID generator request parameter

### ISSUE THREAD

**Issue by Legolaszstudio** [fastify/session#62]
### Prerequisites

- [X] I have written a descriptive issue title
- [x] I have searched existing issues to ensure the feature has not already been requested

### 🚀 Feature Proposal

idGenerator should pass the request parameter to the generator function.

### Motivation

Coming from express-session this seems a reasonable feature, as we assign a prefix to users logged in.

### Example

```ts
fastify.register(fastifySession, {
    store: new RedisStore({ client: redisClient }) as any,
    secret: 'abc123', 
    idGenerator: (req) => {
        if (req.authenticated) return `user_${req.userUUID}:${uuidv1()}`;
        else return uuidv1();
    },
    cookie: {
        httpOnly: true,
    },
});
```

**Comment by mcollina** [comment]
Would you like to send a Pull Request to address this issue? Remember to add unit tests.

**Comment by Legolaszstudio** [comment]
I have a lot of things coming up the following weeks, but I will try my best 😀

**Comment by Legolaszstudio** [comment]
Fixed by #64
